src/rustllvm/RustWrapper.cpp

```diff
@@ -1268,6 +1268,44 @@ extern "C" LLVMValueRef LLVMRustBuildMemMove(LLVMBuilderRef B,
 #endif
 }
 
+extern "C" LLVMValueRef LLVMRustBuildElementUnorderedAtomicMemCpy(LLVMBuilderRef B,
+                                                                  LLVMValueRef Dst, unsigned DstAlign,
+                                                                  LLVMValueRef Src, unsigned SrcAlign,
+                                                                  LLVMValueRef Size, uint32_t ElementSize) {
+  return wrap(unwrap(B)->CreateElementUnorderedAtomicMemCpy(
+      unwrap(Dst), DstAlign,
+      unwrap(Src), SrcAlign,
+      unwrap(Size), ElementSize));
+}
+
+extern "C" LLVMValueRef LLVMRustBuildElementUnorderedAtomicMemMove(LLVMBuilderRef B,
+                                                                   LLVMValueRef Dst, unsigned DstAlign,
+                                                                   LLVMValueRef Src, unsigned SrcAlign,
+                                                                   LLVMValueRef Size, uint32_t ElementSize) {
+#if LLVM_VERSION_GE(7, 0)
+  return wrap(unwrap(B)->CreateElementUnorderedAtomicMemMove(
+      unwrap(Dst), DstAlign,
+      unwrap(Src), SrcAlign,
+      unwrap(Size), ElementSize));
+#else
+  // TODO
+  abort();
```

Comment:
LLVM 6.0 doesn't have a method for the memmove or memcpy intrinsics.

Reply:
Could this and the version below return a `nullptr` in this case and then it's handled in rustc with a bug/panic macro?

Reply:
that's a possibility. Since LLVM 6.0 does have the intrinsics, just not the method, another possibility would be to define something similar to `CreateElementUnorderedAtomicMemMove` (maybe even basing it off the implementation in LLVM 7+).

Reply:
Nah I think it's fine to return `nullptr` here as these are unstable and we don't provide a hard guarantee about support for older LLVM (only a guarantee of support for *some* LLVM which we ship). The easiest thing to do would probably be to return null here and it can be expanded later if need be